native/spark-expr/src/conversion_funcs/cast.rs

```diff
@@ -1190,11 +1262,13 @@ fn is_datafusion_spark_compatible(
                 | DataType::Decimal256(_, _)
                 | DataType::Utf8 // note that there can be formatting differences
         ),
-        DataType::Utf8 if allow_incompat => matches!(
+        DataType::Utf8 if allow_incompat => {
+            matches!(to_type, DataType::Binary | DataType::Decimal128(_, _))
```

Comment:
I don't think `Decimal128` should be added here?

Reply:
@andygrove  thank you . Also I found that String -> Binary is marked compatible on the scala side but marked as `Incompatible` here . I will file an issue to fix that issue and fix it in a separate PR

Reply:
What are your thoughts ?